### ISSUE TITLE
Can't build under CentOS

### ISSUE THREAD

**Issue by jcstanaway** [bloomberg/powerfulseal#206]
For various reasons, our company requires the use of a specific base image - currently CentOS 7.7.

I can `pip3 install powerfulseal` and it appears to run without issue.

However, if I replace that with:
```
WORKDIR /powerfulseal
COPY Makefile LICENSE README.md setup.py setup.cfg MANIFEST.in /powerfulseal/
COPY powerfulseal/ /powerfulseal/powerfulseal/

# install it with pip
RUN pip3 install .
```

it installs, but running it fails:

```
[root@afdc43b6fe05 powerfulseal]# powerfulseal --version
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 584, in _build_master
    ws.require(__requires__)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 901, in require
    needed = self.resolve(parse_requirements(requirements))
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (PyYAML 5.3.1 (/usr/local/lib64/python3.6/site-packages), Requirement.parse('pyyaml~=3.12'), {'kubernetes'})

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/bin/powerfulseal", line 6, in <module>
    from pkg_resources import load_entry_point
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3258, in <module>
    @_call_aside
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3242, in _call_aside
    f(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 3271, in _initialize_master_working_set
    working_set = WorkingSet._build_master()
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 586, in _build_master
    return cls._build_from_requirements(__requires__)
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 599, in _build_from_requirements
    dists = ws.resolve(reqs, Environment())
  File "/usr/local/lib/python3.6/site-packages/pkg_resources/__init__.py", line 792, in resolve
    raise VersionConflict(dist, req).with_context(dependent_req)
pkg_resources.ContextualVersionConflict: (PyYAML 5.3.1 (/usr/local/lib64/python3.6/site-packages), Requirement.parse('pyyaml~=3.12'), {'kubernetes'})
```

I've simplified my dockerfile to just the following:
```
FROM centos:centos7

RUN yum -y update
RUN yum -y install python3

# copy powerfulseal package
WORKDIR /powerfulseal
COPY Makefile LICENSE README.md setup.py setup.cfg MANIFEST.in /powerfulseal/
COPY powerfulseal/ /powerfulseal/powerfulseal/

# install powerfulseal with pip
RUN pip3 install --upgrade .

ENTRYPOINT [ "powerfulseal" ]
```

I'm wanting to pull in the changes described in #205 and so that's the reason for making it from source.

Doing the same, but with `python:3.7-slim-buster` (and using `apt-get`) as the base image works with no issue.

Any pointers would be appreciated.  TIA!

**Comment by dgzlopes** [comment]
`kubernetes` 10.1.0 ties the `pyyaml` version to 3.12. Updating to 11.0.0 fixes the problem. 

In the `setup.cfg` change
```
kubernetes>=10.0.1,<11.0.0
```
To
```
kubernetes>=11.0.0
```
 
(Another option is to explicitly use the older PYAML version, changing `PyYAML>=5.1.2,<6` to `PyYAML==3.12`)

**Comment by seeker89** [comment]
I think that it should be safe to update to `kubernetes>=11.0.0,<12.0.0`. @dgzlopes sounds like you've got that tested locally, would you like to push a PR upstream? Bumping to version 2.9.0.

**Comment by seeker89** [comment]
https://github.com/bloomberg/powerfulseal/pull/207 is merged in, `2.9.0` is building as we speak.

**Comment by jcstanaway** [comment]
Verified that building with latest source (with #207) works.  Closing issue.
